cmd/clusterctl/client/config/reader_viper.go

```diff
@@ -58,14 +58,18 @@ func injectConfigPaths(configPaths []string) viperReaderOption {
 }
 
 // newViperReader returns a viperReader.
-func newViperReader(opts ...viperReaderOption) Reader {
+func newViperReader(opts ...viperReaderOption) (Reader, error) {
+	configDirectory, err := xdg.ConfigFile(ConfigFolder)
+	if err != nil {
+		return nil, err
+	}
 	vr := &viperReader{
-		configPaths: []string{filepath.Join(homedir.HomeDir(), ConfigFolder)},
+		configPaths: []string{configDirectory, filepath.Join(xdg.Home, ConfigFolder)},
```

Comment:
I tested this with a config file in the olf location and its not getting picked up. In the logs i see:

```
No default config file available
```

I think what you need it the following change:

```go
const (
	// ConfigFolder defines the name of the config folder under $XDG_CONFIG_HOME.
	ConfigFolder = "cluster-api"
	// ConfigFolderLegacy defines the name of the old config folder under $HOME.
	ConfigFolderLegacy = ".cluster-api"
....
	vr := &viperReader{
		configPaths: []string{configDirectory, filepath.Join(xdg.Home, ConfigFolderLegacy)},
	}

```

With that change it picks up the file in the old location:

```
Using configuration File="/home/richard/.cluster-api/clusterctl.yaml"
```